src/main/java/uk/ac/cam/cl/dtg/segue/api/userAlerts/UserAlertsWebSocket.java

```diff
@@ -25,123 +25,188 @@
+    /**
+     * Handles a new client websocket connection
+     *
+     * @param session
+     *          - contains information about the session to be started
+     * @throws IOException
+     * @throws SegueDatabaseException
+     * @throws NoUserException
+     * @throws InvalidSessionException
+     */
+    @OnWebSocketConnect
+    public void onConnect(final Session session) throws IOException, SegueDatabaseException, NoUserException, InvalidSessionException {
+        this.session = session;
 
-            while (iter.hasNext())
-                userAlerts.recordAlertEvent(iter.next().asLong(), IUserAlert.AlertEvents.SEEN);
+        Map<String, String> sessionInformation = getSessionInformation(session);
 
+        if (userManager.isValidUserFromSession(sessionInformation)) {
 
-            Map<String, Object> eventDetails = new ImmutableMap.Builder<String, Object>()
-                    .put("notification_ids", alertFeedback.path("notificationIds")).build();
+            connectedUser = userManager.getUserDTOById(Long.parseLong(sessionInformation.get(SESSION_USER_ID)));
 
+            if (!connectedSockets.containsKey(connectedUser.getId())) {
+                connectedSockets.put(connectedUser.getId(), new LinkedList<>());
+            }
+            connectedSockets.get(connectedUser.getId()).add(this);
 
-            logManager.logInternalEvent(connectedUser, NOTIFICATION_VIEW_LIST, eventDetails);
+            // For now, we hijack this websocket class to deliver user streak information
+            sendUserSnapshotData();
 
-        } else if (feedbackType.equals(IUserAlert.AlertEvents.CLICKED.name())) {
+            // TODO: Send initial set of notifications.
+            List<IUserAlert> persistedAlerts = userAlerts.getUserAlerts(connectedUser.getId());
+            if (!persistedAlerts.isEmpty()) {
+                session.getRemote().sendString(objectMapper.writeValueAsString(ImmutableMap.of("notifications", persistedAlerts)));
+            }
 
-            userAlerts.recordAlertEvent(alertFeedback.path("notificationId").asLong(), IUserAlert.AlertEvents.CLICKED);
+        }
+    }
 
-            Map<String, Object> eventDetails = new ImmutableMap.Builder<String, Object>()
-                    .put("notification_id", alertFeedback.path("notificationId").asLong()).build();
 
-            logManager.logInternalEvent(connectedUser, NOTIFICATION_CLICK, eventDetails);
+    /**
+     * Handles the closing of the websocket conenction
+     *
+     * @param session
+     **          - contains information on the currently connected session
+     * @param status
+     *          - the status of the connection
+     * @param reason
+     *          - states the cause for closing the connection
+     */
+    @OnWebSocketClose
+    public void onClose(Session session, int status, String reason) {
+        connectedSockets.get(connectedUser.getId()).remove(this);
```

Comment:
Might want to consider removing the connected user from connectedSockets map altogether, but, in the worst case, will only get to the low tens of thousands of entries given our current release cycle length and number of unique users.